internal/directives/git_commit_directive.go

```diff
@@ -32,22 +35,116 @@ func (g *gitCommitDirective) Name() string {
 
 // Run implements the Directive interface.
 func (g *gitCommitDirective) Run(
+	ctx context.Context,
+	stepCtx *StepContext,
+) (Result, error) {
+	if err := g.validate(stepCtx.Config); err != nil {
+		return Result{Status: StatusFailure}, err
+	}
+	cfg, err := configToStruct[GitCommitConfig](stepCtx.Config)
+	if err != nil {
+		return Result{Status: StatusFailure},
+			fmt.Errorf("could not convert config into %s config: %w", g.Name(), err)
+	}
+	return g.run(ctx, stepCtx, cfg)
+}
+
+// validate validates the git-commit directive configuration against the JSON
+// schema.
+func (g *gitCommitDirective) validate(cfg Config) error {
+	return validate(g.schemaLoader, gojsonschema.NewGoLoader(cfg), g.Name())
+}
+
+func (g *gitCommitDirective) run(
 	_ context.Context,
 	stepCtx *StepContext,
+	cfg GitCommitConfig,
 ) (Result, error) {
-	failure := Result{Status: StatusFailure}
-	// Validate the configuration against the JSON Schema
-	if err := validate(
-		g.schemaLoader,
-		gojsonschema.NewGoLoader(stepCtx.Config),
-		"git-commit",
-	); err != nil {
-		return failure, err
-	}
-	if _, err := configToStruct[GitCommitConfig](stepCtx.Config); err != nil {
-		return failure,
-			fmt.Errorf("could not convert config into git-commit config: %w", err)
-	}
-	// TODO: Add implementation here
+	path, err := securejoin.SecureJoin(stepCtx.WorkDir, cfg.Path)
+	if err != nil {
+		return Result{Status: StatusFailure}, fmt.Errorf(
+			"error joining path %s with work dir %s: %w",
+			cfg.Path, stepCtx.WorkDir, err,
+		)
+	}
+	workTree, err := git.LoadWorkTree(path, nil)
+	if err != nil {
+		return Result{Status: StatusFailure},
+			fmt.Errorf("error loading working tree from %s: %w", cfg.Path, err)
+	}
+	if err = workTree.AddAll(); err != nil {
+		return Result{Status: StatusFailure},
+			fmt.Errorf("error adding all changes to working tree: %w", err)
+	}
+	commitMsg, err := buildCommitMessage(stepCtx.SharedState, cfg)
+	if err != nil {
+		return Result{Status: StatusFailure},
+			fmt.Errorf("error building commit message: %w", err)
+	}
+	commitOpts := &git.CommitOptions{}
+	if cfg.Author != nil {
+		commitOpts.Author = &git.User{}
+		if cfg.Author.Name != "" {
+			commitOpts.Author.Name = cfg.Author.Name
+		}
+		if cfg.Author.Email != "" {
+			commitOpts.Author.Email = cfg.Author.Email
+		}
+	}
+	if err = workTree.Commit(commitMsg, commitOpts); err != nil {
+		return Result{Status: StatusFailure},
+			fmt.Errorf("error committing to working tree: %w", err)
+	}
 	return Result{Status: StatusSuccess}, nil
 }
+
+func buildCommitMessage(sharedState State, cfg GitCommitConfig) (string, error) {
```

Comment:
I would probably make this a method to ensure we do not get it mixed up with the `generateCommitMessage` methods of the directives which _produce_ Git commit messages as part of their output.

Reply:
Fixed in e62495c6b3ec78dc180ffaaff04a8cfdf819a3ad